### PR TITLE
Fix: Check bazel version before trying to setup the workspace.

### PR DESCRIPTION
**Pull request by gunan**
I could not prevent all the other errors from showing up, but this way we at least get the bazel version error message at the end:

```
...
ERROR: /tensorflow/tensorflow/workspace.bzl:351:3: //external:zlib_archive: no such attribute 'urls' in 'new_http_archive' rule.
ERROR: /tensorflow/tensorflow/workspace.bzl:351:3: //external:zlib_archive: missing value for mandatory attribute 'url' in 'new_http_archive' rule.
ERROR: /tensorflow/WORKSPACE:6:1: Traceback (most recent call last):
	File "/tensorflow/WORKSPACE", line 6
		check_version("0.4.2")
	File "/tensorflow/tensorflow/workspace.bzl", line 32, in check_version
		fail("
Current Bazel version is {}, e...))

Current Bazel version is 0.3.2, expected at least 0.4.2
.
ERROR: Error evaluating WORKSPACE file.
ERROR: error loading package 'external': Package 'external' contains errors.
ERROR: missing fetch expression. Type 'bazel help fetch' for syntax and help.
```

### CODE REVIEW

**Comment by vrv** [comment]
Nice, do you know why the order matters?

**Comment by gunan** [comment]
I think before we were blocked by tf_workspace() call below.
When loading the functions, I think they are non-fatal error messages, but just like what check_version does, tf_workspace did break up all evaluation. The errors stem from addition of mirrors to all the dependency rules we have.

File: WORKSPACE
Comment (vrv) [inline WORKSPACE]: Can we be a bit more careful about what is said here?  Maybe we can say "We must check the bazel version before trying to parse any other BUILD files, in case the parsing of those build files depends on the bazel version we require here."

Comment (gunan) [inline WORKSPACE]: Done. Made the change as a separate commit but I can squash all after approval, if you like.

**Review by vrv — APPROVED** [review]

**Review by martinwicke — APPROVED** [review]

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by alexeagle** [comment]
I was copying this for rules_nodejs and rules_typescript and noticed a bug:

`Current Bazel version is 1.12.3, expected at least 1.2.1`

/cc @mprobst

**Comment by Mistobaan** [comment]
I also think that the check_version should be inside the `tf_workspace` function. Is too easy to update tensorflow without updating the check_version constraints.

**Comment by alexeagle** [comment]
FYI, I added unit tests in my clone of this feature:
https://github.com/bazelbuild/rules_nodejs/pull/19/commits/f401ccb6fa79b4b46efe9cfd8cb6a5ae07d6aaa9

**Pull request merged**
